lib/puppet/provider/package/gem.rb

```diff
@@ -92,9 +92,15 @@ def insync?(is)
     is.any? { |version| dependency.match?('', version) }
   end
 
+  def gem_version(cmd)
+    command = [cmd, '--version']
+    execute(command, {:failonfail => true, :combine => true, :custom_environment => {"HOME"=>ENV["HOME"]}})
```

Comment:
IIRC, the `gem` command is sensitive to `HOME` due to `.gemrc`, and this matches the behavior below. The tricky part is if you override any options to `execute`, then you have to also respecify the defaults for `failonfail` and `combine`. It might be good to have a general `gem_command` method that does the option handling, and appends whatever options are passed to it, eg `--version` in this case or the array of arguments on line 147.

Reply:
I think I follow, I’ll try that tomorrow.

Reply:
Added a second commit implementing a general `execute_gem_command` method.